src/main/java/com/google/sps/servlets/QueryFactory.java

```diff
@@ -0,0 +1,53 @@
+package com.google.sps.servlets; 
+
+import com.google.cloud.spanner.Statement;
+import java.util.List;
+
+final class QueryFactory {
+  
+  private static final QueryFactory instance = new QueryFactory();
+  private static final String GROUP_BY_TABLE_NAMES = ") group by table_name";
+
+  private QueryFactory() {
+    // Private because static class
+  }
+
+  static QueryFactory getInstance() {
+    return instance;
+  }
+
+  //TODO: try to do binding w/ parameters
+  static Statement buildSchemaQuery(String table) {
+    StringBuilder query = new StringBuilder("SELECT column_name, spanner_type, is_nullable ");
+    query.append("FROM information_schema.columns WHERE table_name = '%s'");
+    return Statement.newBuilder(String.format(query.toString(), table)).build();
+  }
+
+  // Construct SQL statement of form SELECT <columns list> FROM <table>
+  static Statement constructQueryStatement(List<ColumnSchema> columnSchemas, String table) {
+    StringBuilder query = new StringBuilder("SELECT ");
+
+    for (ColumnSchema columnSchema : columnSchemas) {
```

Comment:
Try something like the following, please double check the syntax though. 

String query = String.format("Select @columns FROM %s ", table);
return Statement.newBuilder(query).bind("columns").to(String.join(", ", columnSchemas.stream().map(ColumnSchema::columnName).collect(Collectors.toList()))).build();

So here 
1. we have _query_ showing the whole query structure
2. try to leverage the statement bind with building the table name in the string
3. Using string join to build the list of columns in a string, but since the column name is in ColumnSchema class, use Java stream to unwrap and collect the names. Can have a local variable String columns = String.join(", ", columnSchemas.stream().map(ColumnSchema::columnName).collect(Collectors.toList()));

Reply:
I tried to do this, and the stream/map part seems to work: it prints out the correct column names. But then when I run the code I get an error saying Field not found: SingerId (which is the first column in the table).

Reply:
Let's just use String formatting then. It would still be nice if you can avoid the for loop and removing the 
extra space manually with String join. 

String columns = String.join(", ", columnSchemas.stream().map(ColumnSchema::columnName).collect(Collectors.toList()));
String query = String.format("Select %s FROM %s ", columns, table);
return Statement.newBuilder(query).build();

Reply:
Done